src/cow-react/modules/limitOrders/updaters/QuoteUpdater/hooks/useFetchMarketPrice/useHandleResponse.ts

```diff
@@ -28,30 +25,25 @@ export function useHandleResponse() {
           return
         }
 
-        const { buyAmount, sellAmount } = data.quote
+        const { buyAmount: buyAmountRaw, sellAmount: sellAmountRaw, feeAmount: feeAmountRaw } = data.quote
 
         if (!outputCurrency || !inputCurrency) {
           return
         }
 
-        // Adjust for decimals
-        const adjustedBuy = JSBI.multiply(JSBI.BigInt(buyAmount), JSBI.BigInt(10 ** getDecimals(inputCurrency)))
-        const adjustedSell = JSBI.multiply(JSBI.BigInt(sellAmount), JSBI.BigInt(10 ** getDecimals(outputCurrency)))
+        const feeAmount = CurrencyAmount.fromRawAmount(inputCurrency, feeAmountRaw)
+        const sellAmount = CurrencyAmount.fromRawAmount(inputCurrency, sellAmountRaw)
+        const buyAmount = CurrencyAmount.fromRawAmount(outputCurrency, buyAmountRaw)
 
-        // Create execution rate fraction
-        const executionRate = new Fraction(adjustedBuy, adjustedSell)
+        const marketRate = FractionUtils.fractionLikeToFraction(
+          new Price({ baseAmount: sellAmount, quoteAmount: buyAmount })
+        )
```

Comment:
are we deliberately removing the 0.1%. I'm not 100% sure we want to do this

Reply:
Yeah, I'm also not sure.
Let's discuss it